sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -31,6 +31,8 @@ def __init__(
         self._instance_discovery = instance_discovery
         self._authority = normalize_authority(authority) if authority else get_default_authority()
         self._regional_authority = os.environ.get(EnvironmentVariables.AZURE_REGIONAL_AUTHORITY_NAME)
+        if self._regional_authority.lower() in ["tryautodetect", "true"]:
```

Comment:
Maybe do ` os.environ.get(EnvironmentVariables.AZURE_REGIONAL_AUTHORITY_NAME, "")` above to ensure the `lower()` always works?

Reply:
Nvm, better to just check value before lower call as you did in the latest commit.